File: C++11/Hartals.cpp
#include <iostream>

using namespace std;

int main() {
    int n_cases;
    cin >> n_cases;

    while(n_cases > 0) {
        n_cases--;

        int res = 0;

        int n_days, n_parties;
        cin >> n_days;
        cin >> n_parties;

        int hartals[n_parties];
        int hartal;

        for(int i = 0; i < n_parties; i++) {
            cin >> hartal;
            hartals[i] = hartal;
        }

        for(int day = 0; day <= n_days; day++) {

            int hartal_cond = (day%7);
            if (hartal_cond == 6 || hartal_cond == 0) {
                continue;
            } else {
                bool cont_cond = false;
                for (int hartal_pos = 0; hartal_pos < n_parties; hartal_pos++) {
                    if ((day%hartals[hartal_pos]) == 0) {
                        res++;
                        cont_cond = true;
                        break;
                    }
                }
                if (cont_cond) {
                    continue;
                }
            }
        }

        cout << res << "\n";
    }
}

File: Python/Hartals.py

n_cases = int(input())

for s in range(n_cases):

    n_days = int(input())
    n_parties = int(input())
    res = 0
    hartals = []
    for i in range(n_parties):
        hartal = int(input())
        if hartal < n_days:
            hartals.append(hartal)
    
    for day in range(1, n_days+1):
        hartal_cond = (day%7)
        if hartal_cond == 6 or hartal_cond == 0:
            continue
        else:
            cont_cond = False
            for hartal in hartals:
                if day%hartal == 0:
                    res += 1
                    cont_cond = True
                    break
            if cont_cond:
                continue
    
    print(res)

